src/app/model/User.ts

```diff
@@ -57,14 +57,20 @@ class User extends BaseModel {
       });
   }
   /**
+   * In case an user have logged in but his user profile is empty
+   * This can only happens if user decided to login without
+   * going the proper signup channel as firebase allow singup
+   * by login as a default
+   *
    * create a user
    * @param {string} userUid - user
    * @param {object} data- updated data
    */
-  createProfile(userUid: string, data: object) {
-    return this.getCollection("users")
-      .doc(userUid)
-      .set(this.load({ data, uid: userUid }));
+  createProfileIfNotExist(auth: any, profile: any) {
+    if (!auth || !profile) return;
+    if (auth.isLoaded && !auth.isEmpty && profile.isEmpty && profile.isLoaded) {
```

Comment:
I'm not sure this works.
I logged out my auth object and profile object
![image](https://user-images.githubusercontent.com/20975817/81084053-6572a300-8ec3-11ea-8307-5e8de34ab35f.png)

profile still comes through as not empty so this would never get triggered

Reply:
Also could we make this a seperate function?

Theres still some other pages that rely on the original createProfile function

Reply:
ehm, this is for create profile for the very first time, you already have your profile created, if you want to stimulate this, you need to either
1. with simulator, just restart it
2. on stage, remove your profile/user from firestore

yes, I will create both function